operator/pkg/ztunnel/controller.go

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ztunnel
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+
+	operatorOption "github.com/cilium/cilium/operator/option"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+const (
+	ztunnelDaemonSetName = "ztunnel-cilium"
+)
+
+// controller manages a ztunnel daemonset, ensuring a ztunnel proxy runs on each
+// node in the cluster.
+type controller struct {
+	client         k8sClient.Clientset
+	logger         *slog.Logger
+	config         Config
+	operatorConfig *operatorOption.OperatorConfig
+}
+
+// create will create the ztunnel daemonset.
+func (c *controller) create(ctx context.Context, ds *appsv1.DaemonSet, ns string) error {
+	_, err := c.client.AppsV1().DaemonSets(ns).Create(ctx, ds, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create ztunnel DaemonSet: %w", err)
+	}
+
+	c.logger.Info("Successfully created ZTunnel DaemonSet")
```

Comment:
Different casing exists around these statements. ZTunnel and ztunnel. Can we ensure these are uniform?